homeassistant/components/daikin/__init__.py

```diff
@@ -4,112 +4,105 @@
 For more details about this component, please refer to the documentation
 https://home-assistant.io/components/daikin/
 """
-import logging
+import asyncio
 from datetime import timedelta
+import logging
 from socket import timeout
 
+import async_timeout
 import voluptuous as vol
 
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOSTS
 import homeassistant.helpers.config_validation as cv
-from homeassistant.components.discovery import SERVICE_DAIKIN
-from homeassistant.const import (
-    CONF_HOSTS, CONF_ICON, CONF_MONITORED_CONDITIONS, CONF_NAME, CONF_TYPE
-)
-from homeassistant.helpers import discovery
-from homeassistant.helpers.discovery import load_platform
+from homeassistant.helpers.typing import HomeAssistantType
 from homeassistant.util import Throttle
 
+from . import config_flow  # noqa  pylint_disable=unused-import
+from .const import KEY_HOST
+
 REQUIREMENTS = ['pydaikin==0.8']
 
 _LOGGER = logging.getLogger(__name__)
 
 DOMAIN = 'daikin'
 
-ATTR_TARGET_TEMPERATURE = 'target_temperature'
-ATTR_INSIDE_TEMPERATURE = 'inside_temperature'
-ATTR_OUTSIDE_TEMPERATURE = 'outside_temperature'
 
 MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
 
 COMPONENT_TYPES = ['climate', 'sensor']
 
-SENSOR_TYPE_TEMPERATURE = 'temperature'
-
-SENSOR_TYPES = {
-    ATTR_INSIDE_TEMPERATURE: {
-        CONF_NAME: 'Inside Temperature',
-        CONF_ICON: 'mdi:thermometer',
-        CONF_TYPE: SENSOR_TYPE_TEMPERATURE
-    },
-    ATTR_OUTSIDE_TEMPERATURE: {
-        CONF_NAME: 'Outside Temperature',
-        CONF_ICON: 'mdi:thermometer',
-        CONF_TYPE: SENSOR_TYPE_TEMPERATURE
-    }
-
-}
-
 CONFIG_SCHEMA = vol.Schema({
     DOMAIN: vol.Schema({
         vol.Optional(
             CONF_HOSTS, default=[]
         ): vol.All(cv.ensure_list, [cv.string]),
-        vol.Optional(
-            CONF_MONITORED_CONDITIONS,
-            default=list(SENSOR_TYPES.keys())
-        ): vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)])
     })
 }, extra=vol.ALLOW_EXTRA)
 
 
 def setup(hass, config):
```

Comment:
We have to change this to a coroutine since we want to be in async context to schedule the forwarding below.